pcsx2/gui/GlobalCommands.cpp

```diff
@@ -150,15 +150,20 @@ namespace Implementations
 			OSDlog( Color_StrongRed, true, "(FrameLimiter) SlowMotion ENABLED." );
 			g_Conf->EmuOptions.GS.FrameLimitEnable = true;
 		}
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
+
 		pauser.AllowResume();
 	}
 
 	void Framelimiter_MasterToggle()
 	{
 		ScopedCoreThreadPause pauser;
 		g_Conf->EmuOptions.GS.FrameLimitEnable = !g_Conf->EmuOptions.GS.FrameLimitEnable;
-		GSsetVsync( g_Conf->EmuOptions.GS.FrameLimitEnable && g_Conf->EmuOptions.GS.VsyncEnable );
 		OSDlog( Color_StrongRed, true, "(FrameLimiter) %s.", g_Conf->EmuOptions.GS.FrameLimitEnable ? "ENABLED" : "DISABLED" );
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
```

Comment:
Not sure I do.

My interpretation: `EmuConfig` is used (or should only be used) by the core emulation code and write access to it is supposed to be restricted to prevent accidental modifications (I note that it being const is likely undefined behaviour). `g_Conf->EmuOptions` is for changes that are supposed to be applied to `EmuConfig`, though it's not always obvious if and when the changes are applied.

I think you have to use `g_Conf->EmuOptions` in this case.